app/models/pubmed_source_record.rb

```diff
@@ -139,22 +171,16 @@ def source_as_hash(publication = Nokogiri::XML(source_data).xpath('//PubmedArtic
     record_as_hash[:author] = author_list.map { |a| author_to_hash(a) }.compact
 
     record_as_hash[:mesh_headings] = mesh_headings if mesh_headings.present?
-    year_xpaths = [
-      'MedlineCitation/Article/Journal/JournalIssue/PubDate/Year',
-      'MedlineCitation/Article/ArticleDate/Year',
-      'PubmedData/History/PubMedPubDate[@PubStatus="accepted"]/Year',
-      'PubmedData/History/PubMedPubDate[@PubStatus="pubmed"]/Year',
-      'PubmedData/History/PubMedPubDate[@PubStatus="medline"]/Year',
-      'PubmedData/History/PubMedPubDate[@PubStatus="entrez"]/Year'
-    ]
-    # look for a year in all of the xpath locations above in order
-    #  stop after the first produces something that looks like a year
-    year_xpaths.each do |path|
-      match = publication.xpath(path).text.match(/[12][0-9]{3}/)
-      next unless match
-
-      record_as_hash[:year] = match.to_s
-      break
+
+    year = extract_year_from_pubmed_record(publication)
+    record_as_hash[:year] = year if year
+
+    month = extract_month_from_pubmed_record(publication)
+    day = extract_day_from_pubmed_record(publication)
+    if year && month && day
+      record_as_hash[:date] =  "#{year}-#{month.rjust(2, '0')}-#{day.rjust(2, '0')}"
+    elsif year && month
+      record_as_hash[:date] =  "#{year}-#{month.rjust(2, '0')}"
     end
```

Comment:
note, we want to 0 pad months and days to be consistent with other date formats, so 2022-1-4 becomes 2022-01-04